atlas-nagios-check/measurements.py

```diff
@@ -0,0 +1,632 @@
+            if args.aa and not self.aa:
+                message.add_error(self.probe_id, 'AA Flag not set')
+            if args.rd and not self.rd:
+                message.add_error(self.probe_id, 'RD Flag not set')
+            if args.ra and not self.ra:
+                message.add_error(self.probe_id, 'RA Flag not set')
+            if args.ad and not self.ad:
+                message.add_error(self.probe_id, 'AD Flag not set')
+            if args.cd and not self.cd:
+                message.add_error(self.probe_id, 'CD Flag not set')
+
+
+class MeasurmentDnsA(MeasurmentDns):
+    '''class for a dns A measuerment'''
+
+    def __init__(self, probe_id, payload):
+        '''Initiate Object'''
+        #super(Measurment, self).__init__(self, payload)
+        MeasurmentDns.__init__(self, probe_id, payload)
+
+    @staticmethod
+    def add_args(subparser):
+        parser = subparser.add_parser('A', help='A DNS check')
+        MeasurmentDns.add_args(parser)
+        parser.add_argument('--answer',
+                help='Ensure the RR set from the answer \
+                        contains a this string can also check if we get a cname')
+
+    def check(self, args, message):
+        if not self.parse_error:
+            MeasurmentDns.check(self, args, message)
+            for answer in self.answers:
+                if answer.type == 'RRSIG':
+                    continue
+                elif answer.type != 'A' and answer.type != 'CNAME':
+                    message.add_error(self.probe_id, self.msg % (
+                        'RRTYPE', answer.type))
+                elif args.answer:
+                    self.check_string( args.answer, answer.address, 'address', message)
+
+class MeasurmentDnsAAAA(MeasurmentDns):
+    '''class for a dns AAAA measuerment'''
+
+    def __init__(self, probe_id, payload):
+        '''Initiate Object'''
+        #super(Measurment, self).__init__(self, payload)
+        MeasurmentDns.__init__(self, probe_id, payload)
+
+    @staticmethod
+    def add_args(subparser):
+        parser = subparser.add_parser('AAAA', help='AAAA DNS check')
+        MeasurmentDns.add_args(parser)
+        parser.add_argument('--answer',
+                help='Ensure the RR set from the answer \
+                        contains a CNAME record with this string')
+
+    def check(self, args, message):
+        if not self.parse_error:
+            MeasurmentDns.check(self, args, message)
+            for answer in self.answers:
+                if answer.type == 'RRSIG':
+                    continue
+                elif answer.type != 'AAAA' and answer.type != 'CNAME':
+                    message.add_error(self.probe_id, self.msg % (
+                        'RRTYPE', answer.type))
+                elif args.answer:
+                    self.check_string( args.answer, answer.address, 'address', message)
+
+class MeasurmentDnsCH(MeasurmentDns):
+    '''class for a dns AAAA measuerment'''
```

Comment:
This comment doesn't match the class name.  Truth be told, all of these comments aren't really necessary since you're using good descriptive class and method names.